### PR TITLE
feat: support register umd plugin as environment plugin

### PR DESCRIPTION
**Pull request by 9aoy**
support register umd plugin as rsbuild environment plugin.

https://rsbuild.dev/plugins/dev/index#environment-plugin

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@rsbuild/plugin-umd",
-	"version": "1.0.1",
+	"version": "1.0.2",
```
Comment (chenjiahan) [inline package.json:3]: we prefer to bump version in a separate PR

File: package.json [hunk 2]
```diff
@@ -34,8 +34,8 @@
 	"devDependencies": {
 		"@biomejs/biome": "^1.8.3",
 		"@playwright/test": "^1.44.1",
-		"@rsbuild/core": "^1.0.1-beta.10",
-		"@rsbuild/plugin-react": "1.0.1-beta.10",
+		"@rsbuild/core": "^1.0.2",
+		"@rsbuild/plugin-react": "1.0.1",
```
Comment (chenjiahan) [inline package.json:38]:
```suggestion
		"@rsbuild/plugin-react": "^1.0.1",
```

File: package.json [hunk 3]
```diff
@@ -46,7 +46,7 @@
 		"typescript": "^5.5.2"
 	},
 	"peerDependencies": {
-		"@rsbuild/core": "1.x || ^1.0.1-beta.0"
+		"@rsbuild/core": "1.x"
```
Comment (chenjiahan) [inline package.json:49]: I think we can remove `^1.0.1-beta.0` in the future, after more users have upgraded to v1 stable.

image

File: package.json (hunk 1 above)
Comment (9aoy) [inline package.json:3]:
```suggestion
	"version": "1.0.1",
```

**Pull request merged**